src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java

```diff
@@ -930,57 +940,38 @@ private void handleThrottleAfterBegin(
         }
     }
 
-    private void handleBoundedWindow(
+    private void handleWindow(
         WindowFW window)
     {
-        final int acceptReplyWindowCredit = window.credit();
+        acceptReplyWindowBudget += window.credit();
         acceptReplyWindowPadding = connectReplyWindowPadding = window.padding();
 
-        acceptReplyWindowBudget += acceptReplyWindowCredit;
-
-        int bufferedDataLength = 0;
         if (slotIndex != NO_SLOT)
         {
             decodeBufferedData();
-            if (slotIndex != NO_SLOT)
-            {
-                bufferedDataLength = slotPosition - slotOffset;
-            }
         }
 
-        final int connectReplyWindowPositiveCredit =
-                Math.max(factory.bufferPool.slotCapacity() - connectReplyWindowBudget - bufferedDataLength, 0);
-
-        connectReplyWindowBudget += connectReplyWindowPositiveCredit;
-
-        if (connectReplyWindowPositiveCredit > 0)
-        {
-            factory.writer.doWindow(connectReplyThrottle, sourceId,
-                                    connectReplyWindowPositiveCredit, connectReplyWindowPadding);
-        }
+        sendWindow();
     }
 
-    private void handleWindow(
-        WindowFW window)
+    private void sendWindow()
     {
-        final int acceptReplyWindowCredit = window.credit();
-        acceptReplyWindowPadding = connectReplyWindowPadding = window.padding();
-
-        acceptReplyWindowBudget += acceptReplyWindowCredit;
-
-        if (slotIndex != NO_SLOT)
+        if (!initialConnectReplyWindowDrained && connectReplyWindowBudget == 0)
         {
-            decodeBufferedData();
+            initialConnectReplyWindowDrained = true;
         }
```

Comment:
Wouldn't it be better to update `connectReplyWindowAligned` to `true` when `connectReplyWindowBudget` is modified and becomes `0` instead of noticing and repetitively testing this condition here?